ProjectHive/ProjectHive.Services.AuthAPI/Services/TokenService.cs

```diff
@@ -0,0 +1,87 @@
+﻿using AutoMapper;
+using Microsoft.IdentityModel.Tokens;
+using ProjectHive.Services.AuthAPI.Data.Entities;
+using ProjectHive.Services.AuthAPI.Data.Repository.Interface;
+using ProjectHive.Services.AuthAPI.Dto;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+using UAParser;
+
+namespace ProjectHive.Services.AuthAPI.Services
+{
+    public class TokenService(IConfiguration configuration, IUserService userService, IUnitOfWork unitOfWork, IMapper mapper) : ITokenService
+    {
+        public async Task<Guid> AddRefreshToken(string email, string userAgent, Guid userId, CancellationToken cancellationToken)
+        {
+            var user = await userService.GetByEmail(email, cancellationToken);
+            if (user != null)
+            {
```

Comment:
GetByEmail return Task<UserDto> - should be Task<UserDto?> or null checking before it has no any sense